arangod/Cluster/ClusterInfo.cpp

```diff
@@ -4008,9 +4007,42 @@ Result ClusterInfo::dropCollectionCoordinator(  // drop collection
   auto replicatedStatesCleanup = futures::Future<Result>{std::in_place};
   if (coll->replicationVersion() == replication::Version::TWO) {
     std::vector<replication2::LogId> stateIds;
-    for (auto pair : VPackObjectIterator(shardsSlice)) {
-      auto shardId = pair.key.copyString();
-      stateIds.emplace_back(LogicalCollection::shardIdToStateId(shardId));
+    velocypack::Slice groupIdSlice = collectionSlice.get("groupId");
```

Comment:
I am looking forward to be able to use Inspectors here, would make this a lot simpler.

We should pair again as soon as this PR is merged, and my PR to add Inspectors for Plan/Collection this can be really simplified \o/

Reply:
Sounds great, looking forward to that!